src/components/Carousel.tsx

```diff
@@ -30,46 +38,68 @@ function Carousel(props: CarouselProps): JSX.Element {
     storage.setItem('slideIdx', slideIdx.toString());
   }, [slideIdx]);
 
+  function goNext(): void {
+    setSlideIdx(old => Math.min(old + 1, props.children.length - 1));
+    props.onNext && props.onNext();
+  }
+
+  function goPrev(): void {
+    setSlideIdx(old => Math.max(old - 1, 0));
+    props.onPrev && props.onPrev();
+  }
+
+  const child = props.children[slideIdx];
+  if (child === undefined && props.children.length > 0) {
+    const slideInBounds = Math.min(Math.max(0, slideIdx), props.children.length - 1);
+    setSlideIdx(slideInBounds);
+  }
+
   return (
-    <div id={'carousel-wrapper'}>
-      { props.title && <h1 id={'title'}>{props.title}</h1> }
-      { props.subtitle && <h2 id={'subtitle'}>{props.subtitle}</h2> }
-      <div id={'carousel'}>
-        <button
-          className={'carousel-btn prev'}
-          style={{
-            visibility: ((props.showPrev !== undefined && props.showPrev) || (slideIdx > 0))
-              ? 'visible'
-              : 'hidden',
-          }}
-          onClick={() => {
-            setSlideIdx(old => Math.max(old - 1, 0));
-            props.onPrev && props.onPrev();
-          }}
-        >
-          <img src={PrevSvg} />
-        </button>
-        <div id={'carousel-content'}>
-          {props.children && props.children.length > 0 && props.children[slideIdx]}
-        </div>
-        <button
-          className={'carousel-btn next'}
-          style = {{
-            visibility:
-              ((props.showNext !== undefined && props.showNext) ||
-               (slideIdx < (props.children ? props.children.length - 1 : 0)))
+    <ChangeSlideContext.Provider value={{next: goNext, prev: goPrev}}>
+      <div id={'carousel-wrapper'}>
+        {props.title && <h1 id={'title'}>{props.title}</h1>}
+        {props.subtitle && <h2 id={'subtitle'}>{props.subtitle}</h2>}
+        <div id={'carousel'}>
+          <button
+            className={'carousel-btn prev'}
+            style={{
+              visibility: child &&
+                ((child.showPrev === undefined && slideIdx > 0) ||
+                  (child.showPrev === true))
                 ? 'visible'
                 : 'hidden',
-          }}
-          onClick={() => {
-            setSlideIdx(old => Math.min(old + 1, props.children ? props.children.length - 1 : 0));
-            props.onNext && props.onNext();
-          }}
-        >
-          <img src={NextSvg} />
-        </button>
+            }}
+            onClick={() => goPrev()}
+          >
+            <img src={PrevSvg} />
+          </button>
+          <div id={'carousel-content'}>
+            {
+              child &&
+              <div id={'carousel-items'}>
+                {child.topText && <h2 id={'body-text'}> {child.topText} </h2>}
+                {child.child}
+                {child.bottomText && <h2 id={'body-text'}> {child.bottomText} </h2>}
+              </div>
+            }
+          </div>
```

Comment:
seems to me that the id carousel-content should be sufficient to encompass all of this! essentially i want to remove unnecessary divs if we dont need them!

I have two ideas on how to take this:

```suggestion
          { child &&
            <div id={'carousel-content'}>
              {child.topText && <h2 id={'body-text'}> {child.topText} </h2>}
              {child.child}
              {child.bottomText && <h2 id={'body-text'}> {child.bottomText} </h2>}
            </div>
          }
```

OR

```suggestion
          <div id={'carousel-content'}>
            { child &&
              <>
                {child.topText && <h2 id={'body-text'}> {child.topText} </h2>}
                {child.child}
                {child.bottomText && <h2 id={'body-text'}> {child.bottomText} </h2>}
              </>
            }
          </div>
```

@arjunr00 need a second opinion on this

Reply:
I agree, plus I think the second one is better just in terms of consistency